deployments/extensions.go

```diff
@@ -0,0 +1,405 @@
+// If the path is relative, join it with the base repository path; if
+// the path is URL download it and return path to downloaded copy
+func (e *Extension) fetchFile(filePath, tmpDir string) (string, error) {
+
+	// 1, local path is absolute, return right away
+	if filepath.IsAbs(filePath) {
+		return filePath, nil
+	}
+
+	name := filepath.Base(filePath)
+	u, err := url.Parse(filePath)
+	if err != nil {
+		return "", err
+	}
+	// 2. full URL, download and return path to copy
+	if u.IsAbs() && u.Host != "" {
+		if err := helpers.DownloadFile(u.String(), name, tmpDir); err != nil {
+			return "", err
+		}
+		return filepath.Join(tmpDir, name), nil
+	}
+	// 3. relative path to extension URL: adapt URL and download
+	u, err = url.Parse(e.Repository)
+	if u.IsAbs() && u.Host != "" {
+		u, _ = u.Parse(e.Name + "/")
+		u, _ = u.Parse(filePath)
+		if err := helpers.DownloadFile(u.String(), name, tmpDir); err != nil {
+			return "", err
+		}
+		return filepath.Join(tmpDir, name), nil
+	}
+	// 4. relative path to extension local path
+	return filepath.Join(e.Repository, e.Name, filePath), nil
+}
+
+func (e *Extension) executeScript(path string) error {
+	tmpDir, err := ioutil.TempDir("", "fuseml-extension-chart")
+	if err != nil {
+		return errors.Wrap(err, "can't create temp directory "+tmpDir)
+	}
+	defer os.RemoveAll(tmpDir)
+
+	fullCmd, err := e.fetchFile(path, tmpDir)
+	if err != nil {
+		return errors.Wrap(err, "failed fetching file from "+path)
+	}
+
+	if out, err := helpers.RunProc(fullCmd, tmpDir, e.Debug); err != nil {
+		return errors.Wrap(err, fmt.Sprintf("Failed running script: %s\n", out))
+	}
+
+	return nil
+}
+
+func (e *Extension) installManifest(path, ns string) error {
+	tmpDir, err := ioutil.TempDir("", "fuseml-extension-chart")
+	if err != nil {
+		return errors.Wrap(err, "can't create temp directory "+tmpDir)
+	}
+	defer os.RemoveAll(tmpDir)
+
+	manifestLocalPath, err := e.fetchFile(path, tmpDir)
+	if err != nil {
+		return errors.Wrap(err, "failed fetching file from "+path)
+	}
+
+	out, err := helpers.Kubectl(fmt.Sprintf("apply --filename %s --namespace %s", manifestLocalPath, ns))
+
+	if err != nil {
+		return errors.Wrap(err, fmt.Sprintf("kubectl apply failed:\n%s", out))
+	}
+	return nil
+}
+
+func (e *Extension) unInstallManifest(path, ns string) error {
+	tmpDir, err := ioutil.TempDir("", "fuseml-extension-chart")
+	if err != nil {
+		return errors.Wrap(err, "can't create temp directory "+tmpDir)
+	}
+	defer os.RemoveAll(tmpDir)
+
+	manifestLocalPath, err := e.fetchFile(path, tmpDir)
+	if err != nil {
+		return errors.Wrap(err, "failed fetching file from "+path)
+	}
+
+	out, err := helpers.Kubectl(fmt.Sprintf("delete --filename %s --namespace %s", manifestLocalPath, ns))
+
+	if err != nil {
+		return errors.Wrap(err, fmt.Sprintf("kubectl delete failed:\n%s", out))
+	}
+	return nil
+}
+
+// TODO move under helpers
+func (e *Extension) installHelmPackage(name, chartPath, ns, valuesPath string) error {
```

Comment:
I would like to suggest naming it `installHelmChart`

Reply:
OK